### PR TITLE
upgrade versions

### PR DESCRIPTION
**Pull request by synarete**
Update smbmetrics to more up-to-date tools and go-lang versions.

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ linters:
   disable-all: true
   enable:
     - asciicheck
-    - deadcode
-    - depguard
+    # - deadcode
+    # - depguard
```
Comment (anoopcs9) [inline .golangci.yaml:9]: Why are we disabling `depguard`?

Comment (synarete) [reply .golangci.yaml]: It is a mistake: with new version of golangci-lint it emitted all kind of errors. It turned out that I just needed to add the relevant configuration in `.golangci.yaml`. Fixed.

File: .golangci.yaml [hunk 2]
```diff
@@ -30,16 +30,23 @@ linters:
...
+    # - varcheck
     - whitespace
 linters-settings:
+  depguard:
+    rules:
+      main:
+        list-mode: lax
+        deny:
+          - pkg: "github.com/sirupsen/logrus"
+            desc: not allowed
```
Comment (anoopcs9) [inline .golangci.yaml:49]: How did you come up with these specific settings? Do we require everything as you added here? I could see some of these from depguard [docs](https://golangci-lint.run/usage/linters/#depguard).

Comment (synarete) [reply .golangci.yaml:49]: This linter has it merit, even if needed this a-bit-confusing configuration. I would prefer to keep it, even with this minor penalty.

Comment (anoopcs9) [reply .golangci.yaml:49]: If so how about adding a comment alongside for `depguard` linter settings in _.golangci.yaml_ to make the intention clear?

Comment (synarete) [reply .golangci.yaml]: Fair enough. Done.

File: .golangci.yaml [hunk 3]
```diff
@@ -30,16 +30,24 @@ linters:
...
     - unused
-    - varcheck
+    # - varcheck
     - whitespace
 linters-settings:
+  depguard:
+    rules:
+      main:
+        list-mode: lax
+        # using explicit 'deny' entry only to suppress linter warns.
```
Comment (anoopcs9) [inline .golangci.yaml:47]:
```suggestion
        # using explicit 'deny' entry only to suppress linter warnings.
```

**Review by anoopcs9 — CHANGES_REQUESTED** [review]
lgtm, see below for a typo.

**Comment by synarete** [comment]
> lgtm, see below for a typo.

Fixed.

**Review by anoopcs9 — APPROVED** [review]

**Pull request merged**